src/test/java/uk/gov/hmcts/reform/sscs/personalisation/ActionFurtherEvidencePersonalisationTest.java

```diff
@@ -0,0 +1,152 @@
+
+    SscsCaseDataWrapper responseWrapper;
+
+    SubscriptionWithType subscriptionWithType;
+
+    Subscriptions subscriptions;
+
+    @BeforeEach
+    void setUp() {
+        openMocks(this);
+
+        when(config.getManageEmailsLink()).thenReturn(Link.builder().linkUrl("http://manageemails.com/mac").build());
+        when(config.getTrackAppealLink()).thenReturn(Link.builder().linkUrl("http://tyalink.com/appeal_id").build());
+        when(config.getEvidenceSubmissionInfoLink()).thenReturn(Link.builder().linkUrl("http://link.com/appeal_id").build());
+        when(config.getManageEmailsLink()).thenReturn(Link.builder().linkUrl("http://link.com/manage-email-notifications/mac").build());
+        when(config.getClaimingExpensesLink()).thenReturn(Link.builder().linkUrl("http://link.com/progress/appeal_id/expenses").build());
+        when(config.getHearingInfoLink()).thenReturn(Link.builder().linkUrl("http://link.com/progress/appeal_id/abouthearing").build());
+        when(config.getOnlineHearingLinkWithEmail()).thenReturn(Link.builder().linkUrl("http://link.com/onlineHearing?email={email}").build());
+        when(macService.generateToken("GLSCRR", PIP.name())).thenReturn("ZYX");
+        when(evidenceProperties.getAddress()).thenReturn(evidencePropertiesAddress);
+        when(hearingContactDateExtractor.extract(any())).thenReturn(Optional.empty());
+        when(notificationDateConverterUtil.toEmailDate(any(LocalDate.class))).thenReturn("1 January 2018");
+
+        Map<String, String> englishMap = new HashMap<>();
+        Map<String, String> welshMap = new HashMap<>();
+
+        Map<LanguagePreference, Map<String, String>> personalisations = new HashMap<>();
+        personalisations.put(LanguagePreference.ENGLISH, englishMap);
+        personalisations.put(LanguagePreference.WELSH, welshMap);
+
+        personalisationConfiguration.setPersonalisation(personalisations);
+
+        SscsCaseData caseData = buildCaseData();
+
+        DynamicList sender = new DynamicList(new DynamicListItem("appellant", "Appellant (or Appointee)"), new ArrayList<>());
+        caseData.setOriginalSender(sender);
+
+        responseWrapper = SscsCaseDataWrapper.builder()
+                .newSscsCaseData(caseData)
+                .build();
+
+        Subscription subscription = Subscription.builder()
+                .tya("GLSCRR")
+                .email("test@email.com")
+                .mobile("07983495065")
+                .subscribeEmail("Yes")
+                .subscribeSms("No")
+                .build();
+
+        subscriptions = Subscriptions.builder().appellantSubscription(subscription).jointPartySubscription(subscription).build();
+
+        subscriptionWithType = new SubscriptionWithType(subscriptions.getAppellantSubscription(), APPELLANT,
+                caseData.getAppeal().getAppellant(), caseData.getAppeal().getAppellant());
+    }
+
+    @ParameterizedTest
+    @MethodSource("furtherEvidenceVariations")
```

Comment:
I have been advised against using MethodSource due to it being not super obvious what is being tested at a glance. Where EnumSource is not possible due to multiple input types being required, then CsvSource is preferred over MethodSource. I understand the reason for this is that it puts the inputs right here at the start of the test.

Reply:
I think it look clear for this case what are parameters of test